cypress/e2e/read-receipts/read-receipts.spec.ts

```diff
@@ -354,4 +357,711 @@ describe("Read receipts", () => {
+                assertUnread(room2, 1);
+            });
+            it("Reading the last threaded message makes the room read", () => {
+                // Given a thread exists
+                goTo(room1);
+                receiveMessages(room2, ["Msg1", threadedOff("Msg1", "Resp1")]);
+                assertUnread(room2, 1);
+                goTo(room2);
+
+                openThread("Msg1");
+                assertRead(room2);
+            });
+            it("Reading a thread message makes the thread read", () => {
+                // Given a thread exists
+                goTo(room1);
+                receiveMessages(room2, ["Msg1", threadedOff("Msg1", "Resp1"), threadedOff("Msg1", "Resp2")]);
+                assertUnread(room2, 2); // (Sanity)
+
+                // When I read the main timeline
+                goTo(room2);
+
+                // Then room does appear unread
+                assertUnread(room2, 2);
+
+                // Until we open the thread
+                openThread("Msg1");
+                assertReadThread("Msg1");
+                assertRead(room2);
+            });
+            it.skip("Reading an older thread message (via permalink) leaves the thread unread", () => {});
+            it("Reading only one thread's message does not make the room read", () => {
+                goTo(room1);
+                receiveMessages(room2, ["Msg1", threadedOff("Msg1", "Resp1"), "Msg2", threadedOff("Msg2", "Resp2")]);
+                assertUnread(room2, 4);
+                goTo(room2);
+                assertUnread(room2, 4);
+
+                openThread("Msg1");
+                assertUnread(room2, 1);
+            });
+            it("Reading only one thread's message make that thread read but not others", () => {
+                goTo(room1);
+                receiveMessages(room2, ["Msg1", "Msg2", threadedOff("Msg1", "Resp1"), threadedOff("Msg2", "Resp2")]);
+                assertUnread(room2, 4); // (Sanity)
+
+                // When I read the main timeline
+                goTo(room2);
+
+                assertUnread(room2, 2);
+                assertUnreadThread("Msg1");
+                assertUnreadThread("Msg2");
+
+                openThread("Msg1");
+                assertReadThread("Msg1");
+                assertUnreadThread("Msg2");
+            });
+            it("Reading the main timeline does not mark a thread message as read", () => {
+                // Given a thread exists
+                goTo(room1);
+                receiveMessages(room2, ["Msg1", threadedOff("Msg1", "Resp1"), threadedOff("Msg1", "Resp2")]);
+                assertUnread(room2, 3); // (Sanity)
+
+                // When I read the main timeline
+                goTo(room2);
+
+                assertUnread(room2, 2);
+                // Then thread does appear unread
+                assertUnreadThread("Msg1");
+            });
+            // XXX: this failure seems legit
```

Comment:
Why? If there are new messages in a thread and then I mark the room as read, I think the room should be read.

Reply:
Hang on, I now think I understand your comment. I agree that this failure is legit i.e. the test is failing because our code is not working - the room remains "bold". I added more detail to the comment.